### PR TITLE
Better handling of SQLite error messages

### PR DESCRIPTION
**Pull request by riverszhang89**
Consolidate comdb2 custom error messages into SQLite core. This may also address some of the "unknown error" cases we have seen.

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/317a031e/smoketest.log'>Error &#x26A0; </a>.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>8/400 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>comdb2sys_pagesize_generated [setup failure]<br>sc_timepart_logicalsc_generated<br>yast<br>datetime_mifid2<br>blob_size_limit<br>datetime<br>cdb2api<br>logfill</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/317a031e/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/317a031e/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2729'>Resubmit</a></p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>1/400 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_timepart</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/99cb9487/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/99cb9487/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2729'>Resubmit</a></p>

File: sqlite/src/main.c [hunk 1]
```diff
@@ -1549,6 +1569,14 @@ const char *sqlite3ErrStr(int rc){
...
         zErr = aMsg[rc];
       }
+#if defined(SQLITE_BUILDING_FOR_COMDB2)
+      else if( rc>=comdb2MinErr ){
+          rc -= comdb2MinErr;
+          if( rc<ArraySize(aComdb2Msg) && aComdb2Msg[rc]!=0 ){
+            zErr = aComdb2Msg[rc];
+          }
+      }
+#endif
```
Comment (mistachkin) [inline sqlite/src/main.c:1579]: Minor nit: This `#endif` and the one above it are missing the special comment marker, e.g.:

`#endif /* defined(SQLITE_BUILDING_FOR_COMDB2) */`

Comment (riverszhang89) [reply sqlite/src/main.c]: Yes. Thank you, Joe!

**Review by adizaimi — APPROVED** [review]
Looks good!

**Review by adizaimi — APPROVED** [review]
LGTM!

**Pull request merged**